packages/data/addon/adapters/fact-interface.ts

```diff
@@ -58,7 +60,29 @@ export type RequestV2 = {
   requestVersion: '2.0';
 };
 
+export enum QueryStatus {
+  COMPLETE = 'complete',
+  QUEUED = 'queued',
+  PROCESSING = 'processing',
+  CANCELLED = 'cancelled',
+  TIMEDOUT = 'timedout',
+  FAILURE = 'failure'
+}
+
+export interface AsyncQuery {
+  requestId: string;
+  principalName: string;
+  request: RequestV1 | RequestV2;
+  status: QueryStatus;
+  result: NaviFactsModel;
```

Comment:
This should be 
```suggestion
  result: AsyncQueryResult;
```
based on the RFC

Reply:
Right, this is for the short term. Need to come back.